h2o-core/src/main/java/hex/Model.java

```diff
@@ -2679,6 +2680,24 @@ private InteractionPair(int v1, int v2, String[] v1Enums, String[] v2Enums) {
     }
   }
 
+    /**
+     * Uploads a binary model from a given frame.
+     * Note: binary model has to be created by the same version of H2O, import of a model from a different version will fail
+     * @param destinationFrame key of the frame containing the binary representation of the model on a local filesystem, HDFS, S3...
+     * @return instance of an H2O Model
+     * @throws IOException when reading fails
+     */
+    public static <M extends Model<?, ?, ?>> M uploadBinaryModel(String destinationFrame) throws IOException {
+      Frame fr = DKV.getGet(destinationFrame);
+      try (VecDataInputStream vecDataInputStream = new VecDataInputStream(fr.vec(0))) {
```

Comment:
I think VecDataInputStream is an overkill in this case, not needed. Please use vec.openStream() instead

Reply:
fixed